src/librustc_resolve/lib.rs

```diff
@@ -708,7 +708,6 @@ impl<'a, 'tcx> Visitor<'tcx> for Resolver<'a> {
                 ItemRibKind
             }
             FnKind::Method(_, sig, _, _) => {
-                self.visit_generics(&sig.generics);
```

Comment:
I'm worried about whether this change has any side-effects we don't want. I don't think I found a place to add this back in after I deleted it.

Reply:
Indeed, we will need to add this back somewhere =) I think the problem is that the `FnKind` enum has to change. Methods should operate more analogously with `ItemFn` -- so we should add a `&'a Generics` to [the `Method` variant](https://github.com/rust-lang/rust/blob/0962b8fe4ffc1face685bc1b04364f6a3f0ad1ea/src/libsyntax/visit.rs#L37), and then when we construct `FnKind::Method` ([here](https://github.com/rust-lang/rust/blob/0962b8fe4ffc1face685bc1b04364f6a3f0ad1ea/src/libsyntax/visit.rs#L571) and [here](https://github.com/rust-lang/rust/blob/0962b8fe4ffc1face685bc1b04364f6a3f0ad1ea/src/libsyntax/visit.rs#L594)) we can add the data from the trait or impl item respectively. Then we can restore the call to `visit_generics` that occurs *here*, I suppose.

Reply:
Well, let me read a bit more into this actually. I think what I said is not wrong but a few more tweaks are likely needed.

Reply:
Yeah, so I think it's fine to remove the call here -- it is being *moved* into the `walk_trait_item` and `walk_impl_item` code, effectively.

Reply:
Is this change really okay? If `ItemFn` visits generics, shouldn't `Method` visit them too?

Reply:
Nope. It happens as part of `visit_trait_item` or `visit_impl_item`, I think.